### PR TITLE
use easynetq to handle rabbit mq connections and help with management

### LINKED ISSUE #105: RabbitMQ interface
As part of the processing, services will interact with the RabbitMQ queue:
- the user service will produce email requests to the email service via RabbitMQ queue
- the notification service will produce email requests to the email service via RabbitMQ queue
- the email service will consume requests off the queue and send emails

We need a wrapper object over the RabbitMQ client that will
- read the config for the hostname and all the other parameters required to get a connection to the queue
- will keep a connection pool
- will provide interface for consuming from/producing to a queue

### PR DESCRIPTION
**Pull request by idormenco**
closes #105

### CODE REVIEW

File: backend/src/StamAcasa.Common/Queue/QueueService.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        private const string EmailRequestsQueName = "email:requests";
+        private readonly IBus _bus;
+
+        private readonly Dictionary<string, IQueue> _availableQueues = new Dictionary<string, IQueue>();
+        public QueueService(IBus bus)
+        {
+            _bus = bus;
+            // register queues
+            IQueue queueDeclare = _bus.Advanced.QueueDeclare(EmailRequestsQueName);
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Queue/QueueService.cs]: Small typo
```suggestion
            IQueue queueDeclare = _bus.Advanced.QueueDeclare(EmailRequestsQueueName);
```

File: backend/src/StamAcasa.Common/Queue/QueueService.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EasyNetQ;
+using EasyNetQ.Topology;
+
+namespace StamAcasa.Common.Queue
+{
+    public class QueueService : IQueueService
+    {
+        private const string EmailRequestsQueName = "email:requests";
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Queue/QueueService.cs]: Small typo
```suggestion
        private const string EmailRequestsQueueName = "email:requests";
```

File: backend/src/StamAcasa.Common/Queue/QueueService.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        private readonly Dictionary<string, IQueue> _availableQueues = new Dictionary<string, IQueue>();
+        public QueueService(IBus bus)
+        {
+            _bus = bus;
+            // register queues
+            IQueue queueDeclare = _bus.Advanced.QueueDeclare(EmailRequestsQueName);
+
+            // add them to dictionary
+            _availableQueues.Add(EmailRequestsQueName, queueDeclare);
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Queue/QueueService.cs]: Small typo
```suggestion
            _availableQueues.Add(EmailRequestsQueueName, queueDeclare);
```

File: backend/src/StamAcasa.Common/Queue/QueueService.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+        public async Task PublishNotification<T>(T message) where T : class
+        {
+            throw new NotImplementedException("to do");
+        }
+
+        public async Task PublishEmailRequest<T>(T message) where T : class
+        {
+            var messageWrapper = new Message<T>(message);
+            await _bus.Advanced.PublishAsync(Exchange.GetDefault(), EmailRequestsQueName, false, messageWrapper);
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Queue/QueueService.cs]:
```suggestion
            await _bus.Advanced.PublishAsync(Exchange.GetDefault(), EmailRequestsQueueName, false, messageWrapper);
```

File: backend/src/StamAcasa.Common/Queue/QueueService.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+        public void SubscribeConsumerToNotificationsQueue<T>(Func<T, Task> messageHandler) where T : class
+        {
+            throw new NotImplementedException("to do ");
+        }
+
+        public void SubscribeConsumerToEmailRequestsQueue<T>(Func<T, Task> messageHandler) where T : class
+        {
+            var emailQueue = _availableQueues[EmailRequestsQueName];
```
Comment (RaduCStefanescu) [inline backend/src/StamAcasa.Common/Queue/QueueService.cs]:
```suggestion
            var emailQueue = _availableQueues[EmailRequestsQueueName];
```

**Review by RaduCStefanescu — DISMISSED** [review]

**Review by RaduCStefanescu — APPROVED** [review]

**Review by bvizureanu — APPROVED** [review]

**Review by Puiu — APPROVED** [review]

**Pull request merged**